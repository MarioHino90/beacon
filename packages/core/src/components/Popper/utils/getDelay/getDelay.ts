import { DelayArray, DelayType } from "../../types";

export /**
 * Function that returns an array of delay values if delay
 * is defined, null otherwise
 * @param delay an array of two numbers or a single number
 */
const getDelay = (delay: DelayType | undefined): DelayArray | null => {
  if (Array.isArray(delay)) {
    return delay;
  }

  if (typeof delay === "number") {
    return [delay, delay];
  }

  return null;
};
