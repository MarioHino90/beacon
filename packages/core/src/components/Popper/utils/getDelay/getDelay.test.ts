import { getDelay } from "./getDelay";

describe(`getDelay`, () => {
  it(`should match snapshot when 'delay' arg is an array`, () => {
    expect(getDelay([0, 10])).toMatchSnapshot();
  });

  it(`should match snapshot when 'delay' arg is a number`, () => {
    expect(getDelay(10)).toMatchSnapshot();
  });

  it(`should match snapshot when 'delay' arg is undefined`, () => {
    expect(getDelay(undefined)).toMatchSnapshot();
  });

  it(`should match snapshot when 'delay' arg is something else`, () => {
    expect(getDelay("hello" as any)).toMatchSnapshot();
  });
});
