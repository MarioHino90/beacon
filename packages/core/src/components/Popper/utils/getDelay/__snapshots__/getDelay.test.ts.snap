// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDelay should match snapshot when 'delay' arg is a number 1`] = `
[
  10,
  10,
]
`;

exports[`getDelay should match snapshot when 'delay' arg is an array 1`] = `
[
  0,
  10,
]
`;

exports[`getDelay should match snapshot when 'delay' arg is something else 1`] = `null`;

exports[`getDelay should match snapshot when 'delay' arg is undefined 1`] = `null`;
